configure.ac

```diff
@@ -2669,6 +2677,61 @@ AC_SUBST(GOOPT)
 AC_SUBST(GCCGOOPT)
 AC_SUBST(GOVERSIONOPTION)
 
+#----------------------------------------------------------------
+# Look for Fortran
+#----------------------------------------------------------------
+
+AC_ARG_WITH(fortran,
+  AS_HELP_STRING([--without-fortran], [Disable Fortran])
+  AS_HELP_STRING([--with-fortran=path], [Set location of Fortran compiler])
+  , [FORTRAN="$withval"], [FORTRAN="$alllang_default"])
+
+
+if test x"${FORTRAN}" = xno; then
+  AC_MSG_NOTICE([Disabling Fortran])
+  FC=
+else
+  if test "x$FORTRAN" = xyes; then
+    AC_PROG_FC
+  else
+    FC=$FORTRAN
+  fi
+
+  if test -n "$FC"; then
+    AC_FC_LIBRARY_LDFLAGS dnl fortran library flags
+  fi
+
+  if test -n "$FCFLAGS"; then
+    FCFLAGS="$FCFLAGS"
+  elif test "x$GCC" = xyes; then
+    # NOTE: unit tests use C preprocessor, so -cpp must be set
+    FCFLAGS="-std=f2003 -Wall -Wextra -Wimplicit-procedure -Wimplicit-interface -Wno-compare-reals -ffree-line-length-none -cpp"
+  else
```

Comment:
- `-Wcompare-reals` is turned on by `-Wextra`, which for gfortran doesn't seem like it adds a lot, so I'll remove those two flags. (A warning is emitted when floating-point numbers are compared for equality. Fortran's treatment of numbers is different than in C: it's free to reorganize operations within a statement, so depending on the compiler the same expression may evaluate to two different floating point values.)
- The `-ffree-line-length` shouldn't be necessary any longer since over the development period I've added code to wrap lines to the default Fortran hard maximum of 128 characters. It can be an issue when using preprocessing macros to expand code, so that might surprise some users...
- The `-std=f2003` flag is more about ensuring that the examples and generated code don't accidentally contain GNU extensions or F2008 code. It's restrictive rather than additive like the `-std=C++11` flag.

I don't have access to any compilers that *don't* support the Fortran 2003 features in use. The one feature that I've had trouble with (the `final` attribute, essentially a destructor for a Fortran class) is optional and has warnings in the manual about using it.

In a perfect world there would be feature detection minicodes like those that exist in [CMake for C++11 features](https://cmake.org/cmake/help/v3.1/prop_gbl/CMAKE_CXX_KNOWN_FEATURES.html#prop_gbl:CMAKE_CXX_KNOWN_FEATURES). But Fortran users, bless them, are [used to spotty support for new features](https://www.fortranplus.co.uk/app/download/23704631/fortran_2003_2008_compiler_support.pdf).